core/src/test/scala/unit/kafka/utils/TestUtils.scala

```diff
@@ -230,7 +230,9 @@ object TestUtils extends Logging {
     if (nodeId >= 0) props.put(KafkaConfig.BrokerIdProp, nodeId.toString)
     props.put(KafkaConfig.ListenersProp, listeners)
     if (logDirCount > 1) {
-      val logDirs = (1 to logDirCount).toList.map(i => TestUtils.tempDir().getAbsolutePath).mkString(",")
+      val logDirs = (1 to logDirCount).toList.map(i =>
```

Comment:
Can we add a comment to explain the reason we are doing this?

Reply:
@becketqin Thanks for the review. I have updated the patch with the comment:

```
// We would like to allow user to specify both relative path and absolute path as log directory for backward-compatibility reason
// We can verify this by using a mixture of relative path and absolute path as log directories in the test
```

Reply:
Actually do we really need the `if (logDirCount > 1) ... else ...` block? Can we simplify it to:
```
val logDirs = (1 to logDirCount).toList.map(i =>
         if (i%2 == 0) TestUtils.tempDir().getAbsolutePath else TestUtils.tempRelativeDir("data")
       ).mkString(",")
props.put(KafkaConfig.LogDirsProp, logDirs)
```

Reply:
@becketqin I tried this in KIP-112 patch. It doesn't work directly because some tests rely on `KafkaConfig.LogDirProp`. I think it may be simpler to use the if/else statement instead of having to change these tests. I can also update those tests to uniformly use `KafkaConfig.LogDirsProp` if you prefer.

Reply:
Ah, got it. Then we can leave it as is. It would be necessary to test both configurations.